### ISSUE TITLE
Significance of docstore in TestsetGenerator.

### ISSUE THREAD

**Issue by adityamity** [explodinggradients/ragas#1316]
[ ] I checked the [documentation](https://docs.ragas.io/) and related resources and couldn't find an answer to my question.

**Your Question**
What is the use of docstore in TestsetGenerator. How it will affect the evaluation process if use it and if i do not use it

**Code Examples**
```
class TestsetGenerator:
    generator_llm: BaseRagasLLM
    critic_llm: BaseRagasLLM
    embeddings: BaseRagasEmbeddings
    docstore: DocumentStore

    @classmethod
    def from_llama_index(
        cls,
        generator_llm: LlamaindexLLM,
        critic_llm: LlamaindexLLM,
        embeddings: LlamaIndexEmbeddings,
        docstore: t.Optional[DocumentStore] = None,
        run_config: RunConfig = RunConfig(),
    ) -> "TestsetGenerator":
        generator_llm_model = LlamaIndexLLMWrapper(generator_llm)
        critic_llm_model = LlamaIndexLLMWrapper(critic_llm)
        embeddings_model = LlamaIndexEmbeddingsWrapper(embeddings)
        keyphrase_extractor = KeyphraseExtractor(llm=generator_llm_model)
        if docstore is None:
            from langchain.text_splitter import TokenTextSplitter

            splitter = TokenTextSplitter(chunk_size=1024, chunk_overlap=0)
            docstore = InMemoryDocumentStore(
                splitter=splitter,
                embeddings=embeddings_model,
                extractor=keyphrase_extractor,
                run_config=run_config,
            )
        return cls(
            generator_llm=generator_llm_model,
            critic_llm=critic_llm_model,
            embeddings=embeddings_model,
            docstore=docstore,
        )
```

**Additional context**
When I generate the testset using `TestsetGenerator.from_llama_index ` i found that for many of the 'questions' the gound_truth is coming as 'The answer to given question is not present in context'. Please also help me understand from where does Ragas is getting the `ground_truth` and `contexts`?

**Comment by adityamity** [comment]
@dosu  please help me understand why for many of the questions, I am getting 'ground_truth' as '`The answer to given question is not present in context' even though the `context` is available in 'contexts' column.

**Comment by adityamity** [comment]
@dous from where and how does Ragas create '`Contexts`'?

**Comment by adityamity** [comment]
@dosu from where and how does Ragas create 'Contexts' using TestSetGenerator?

**Comment by adityamity** [comment]
@dosu Let me rephrase my question. To evaluate my RAG application build on llamaindex, i am using RAGAS. To evaluate using RAGAS we need to first generate the testset using `TestsetGenerator` with the below code:
```

from ragas.testset.generator import TestsetGenerator
from ragas.testset.evolutions import simple, reasoning, multi_context

generator = TestsetGenerator.from_llama_index(
    generator_llm=generator_llm,
    critic_llm=critic_llm,
    embeddings=embeddings,
)
# generate testset
testset = generator.generate_with_llamaindex_docs(
    documents,
    test_size=5,
    distributions={simple: 0.5, reasoning: 0.25, multi_context: 0.25},
)
```

This testset contains columns `'question', 'contexts', 'ground_truth' ,'evolution_type','metadata','episode_done'.`

When is did the evaluation using the testset using the below code:

```
from ragas.integrations.llama_index import evaluate
ds = testset.to_dataset()

ds_dict = ds.to_dict()
result = evaluate(
    query_engine=query_engine,
    metrics=metrics,
    dataset=ds_dict,
    llm=evaluator_llm,
    embeddings=OpenAIEmbedding(),
)

```

I found that for many questions, the `ground_truth` is coming as '`The answer to given question is not present in context' `but the `answer`  column has a correct answer. Is it RAGAS not able to generate the `ground_truth` properly or the `contexts` are not properly fetched. How to handle such cases?

**Comment by adityamity** [comment]
@dosu  I am using AzureAISearchVectorStore , what changes i need to do to handle the above scenario ?

**Comment by anistark** [comment]
docstore no longer exists in recent versions.
Checkout KnowledgeGraph and latest docs for current usage.
